File: packages/python/palpluss/types/channels.py
from __future__ import annotations

from typing_extensions import TypedDict

from .common import ChannelType


class PaymentWalletChannel(TypedDict):
    id: str
    tenantId: str
    category: str
    type: ChannelType
    shortcode: str
    name: str
    accountNumber: str | None
    isDefault: bool
    createdAt: str

File: src/types/channels.ts
import type { ChannelType } from './common.js';

export interface PaymentWalletChannel {
  id: string;
  tenantId: string;
  category: 'PAYMENT_WALLET';
  type: ChannelType;
  shortcode: string;
  name: string;
  accountNumber: string | null;
  isDefault: boolean;
  createdAt: string;
}

export interface CreateChannelRequest {
  type: 'PAYBILL' | 'TILL' | 'SHORTCODE';
  shortcode: string;
  name: string;
  accountNumber?: string;
  isDefault?: boolean;
}

export interface UpdateChannelRequest {
  type?: 'PAYBILL' | 'TILL' | 'SHORTCODE';
  shortcode?: string;
  name?: string;
  accountNumber?: string;
  isDefault?: boolean;
}
